### PR TITLE
Add typing to concrete taskflow examples

### PR DESCRIPTION
**Pull request by jscheffl**
I just saw the PR #33296 in Slack and really liked it - I was struggeling to find concrete examples myself multiple times. Thanks to  @RNHTTR for making this!

When taking a look to the examples I noticed that they were not using any typing. I really much like Python type annotations to check code and preventing typos, allowing mypy also to check DAG code.

This PR is a proposal - other opinions welcome - to add typing to examples.

### CODE REVIEW

**Comment by RNHTTR** [comment]
I thought about this, but I was concerned that including just the imports for type hinting in the examples but not including _all_ imports and boilerplate (e.g. `@dag(...)` might be confusing for newer users. With that said, I'm not necessarily opposed, and it's probably for the best to spread the word of type hinting/checking in Python!

**Review by RNHTTR — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

File: docs/shared/template-examples/taskflow.rst
Comment (uranusjr) [inline docs/shared/template-examples/taskflow.rst:25]: I’ve had a mind to add a fix to the `=None` default (they really shouldn’t be needed), maybe I should do it now.

**Review by uranusjr — APPROVED** [review]

**Comment by RNHTTR** [comment]
I think this might have been reverted somehow?

https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/taskflow.html#context

**Comment by uranusjr** [comment]
The change is present in main, but it looks like this entire change didn’t make the 2.7 branch. Not sure if it’s intended.

**Comment by RNHTTR** [comment]
> The change is present in main, but it looks like the entire taskflow documentation change (including this PR and some older rewrites it depends on) didn’t make the 2.7 branch. Not sure if it’s intended.

Odd. I coulda sworn I referred to this piece after this was merged. Do you know if it'd be possible to get this into the next 2.7 release?

**Comment by potiuk** [comment]
This is doc-only so it **should** be possible. The approach is - you mark it with upcoming milestone and hope it will be easy for the release manager to cherry-pick

**Comment by potiuk** [comment]
cc: @ephraimbuddy

**Pull request merged**
